### ISSUE TITLE
Deploying os_server with additional_volumes fails during schema validator checks

### ISSUE THREAD

**Issue by Dannyb48** [CentOS-PaaS-SIG/linchpin#1137]
**Describe the bug**
Using linchpin 1.7.4.1 I'm trying to deploy an OpenStack instance with additional volumes using the workspace example 

https://github.com/CentOS-PaaS-SIG/linchpin/tree/develop/docs/source/examples/workspaces/os-server-addl-vols

PinFile contains the following:

```
--
os-server-addl-vols:
  topology:
    topology_name: os-server-addl-vols
    resource_groups:
      - resource_group_name: os-server-addl-vols
        resource_group_type: openstack
        resource_definitions:
          - name: "db2-linchpin"
            role: os_server
            flavor: m1.small
            image: CentOS-7-x86_64-GenericCloud-1612
            count: 2
            keypair: db2-test
            additional_volumes:
              - name: "volume"
                size: 1
                device_name: "/dev/vdb"
            networks:
              - provider_net_cci_4
        credentials:
          filename: clouds.yaml
          profile: ci-rhos
  layout:
    inventory_layout:
      vars:
        hostname: __IP__
      hosts:
        addl-vols-node:
          count: 2
          host_groups:
            - db2-linchpin
```

The error I get is 
```
 linchpin --creds-path /home/dbaez/projects/carbon/sample_test/linchpin/workspaces/os-server-addl-vols/.  -v up
/home/dbaez/.virtualenvs/carbon/lib/python2.7/site-packages/linchpin/utils/dataparser.py:99: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  data = yaml.load(data)
Traceback (most recent call last):
  File "/home/dbaez/.virtualenvs/carbon/bin/linchpin", line 11, in <module>
    sys.exit(runcli())
  File "/home/dbaez/.virtualenvs/carbon/lib/python2.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/home/dbaez/.virtualenvs/carbon/lib/python2.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/home/dbaez/.virtualenvs/carbon/lib/python2.7/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/dbaez/.virtualenvs/carbon/lib/python2.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/dbaez/.virtualenvs/carbon/lib/python2.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/home/dbaez/.virtualenvs/carbon/lib/python2.7/site-packages/click/decorators.py", line 64, in new_func
    return ctx.invoke(f, obj, *args, **kwargs)
  File "/home/dbaez/.virtualenvs/carbon/lib/python2.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/home/dbaez/.virtualenvs/carbon/lib/python2.7/site-packages/linchpin/shell/__init__.py", line 292, in up
    inv_f=inventory_format)
  File "/home/dbaez/.virtualenvs/carbon/lib/python2.7/site-packages/linchpin/cli/__init__.py", line 391, in lp_up
    tx_id=tx_id)
  File "/home/dbaez/.virtualenvs/carbon/lib/python2.7/site-packages/linchpin/cli/__init__.py", line 578, in _execute_action
    run_id=run_id)
  File "/home/dbaez/.virtualenvs/carbon/lib/python2.7/site-packages/linchpin/cli/__init__.py", line 753, in _execute
    tx_id=tx_id)
  File "/home/dbaez/.virtualenvs/carbon/lib/python2.7/site-packages/linchpin/__init__.py", line 556, in do_action
    resources = validator.validate(provision_data[target])
  File "/home/dbaez/.virtualenvs/carbon/lib/python2.7/site-packages/linchpin/validator/__init__.py", line 49, in validate
    resources = self.validate_topology(topo_data)
  File "/home/dbaez/.virtualenvs/carbon/lib/python2.7/site-packages/linchpin/validator/__init__.py", line 150, in validate_topology
    self.validate_resource_group(group)
  File "/home/dbaez/.virtualenvs/carbon/lib/python2.7/site-packages/linchpin/validator/__init__.py", line 217, in validate_resource_group
    if not v.validate(document):
  File "/home/dbaez/.virtualenvs/carbon/lib/python2.7/site-packages/cerberus/validator.py", line 997, in validate
  
  File "/home/dbaez/.virtualenvs/carbon/lib/python2.7/site-packages/cerberus/validator.py", line 997, in validate
    self.__validate_definitions(definitions, field)
  File "/home/dbaez/.virtualenvs/carbon/lib/python2.7/site-packages/cerberus/validator.py", line 1066, in __validate_definitions
    result = validate_rule(rule)
  File "/home/dbaez/.virtualenvs/carbon/lib/python2.7/site-packages/cerberus/validator.py", line 1040, in validate_rule
    validator = self.__get_rule_handler('validate', rule)
  File "/home/dbaez/.virtualenvs/carbon/lib/python2.7/site-packages/cerberus/validator.py", line 362, in __get_rule_handler
    "domain.".format(rule, domain)
RuntimeError: There's no handler for 'device_name' in the 'validate' domain.
```
If I uninstall linchpin **1.7.4.1** and install **1.6.0**, when the *additional_volumes* param was introduced. It works. The difference looks to be that in 1.6.0 the **Cerberus** package was @ 1.1, it is now @ 1.3.1.

I think the openstack schema.json needs to be updated in the following manner:

```
"additional_volumes": {
                        "type": "list",
                        "required": false,
                        "schema": {
                            "type": "dict",
                            "schema": {
                              "name": { "type": "string", "required": true },
                              "size": { "type": "integer", "required": false },
                              "device_name": { "type": "string", "required": false }
                            }
                        }

```

**Comment by samvarankashyap** [comment]
Hi, @Dannyb48 Thankyou for reporting the issue. 
Since it is a very small fix are you willing to contribute a fix for the current issue? As we are a small team we are currently working with other issues which came up. It might take time by the time we try to fix this bug. 
However, We highly encourage external contributors and try to merge their changes as soon as possible. 
Please let me know if you are interested. 
Thank you

**Comment by samvarankashyap** [comment]
@Dannyb48 : Is this issue fixed with the recent PR #1160 ??

**Comment by Dannyb48** [comment]
Yes.
